res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -2253,358 +1888,222 @@ NumarkMixtrack3.PFLButton = function(channel, control, value, status, group) {
-        }
+        engine.setParameter("[EffectRack1_EffectUnit" + deck.decknum + "]", "mix", mixValue + increment);
     }
 
     if (deck.PADMode) {
-        // PadMode button is used, Shift key is not used, we adjust Sampler volumes
-        // Define new value of sampler pregain
-        // Off = 1, centered = 1, max = 4
-
-        var startingSampler;
-        if (decknum === 1 || decknum === 3) {
-            startingSampler = 1;
+        if (increase) {
+            engine.setValue(deck.group, "beats_translate_later", true);
         } else {
-            startingSampler = 5;
+            engine.setValue(deck.group, "beats_translate_earlier", true);
         }
+    }
 
-        for (i = startingSampler; i <= startingSampler + 3; i++) {
-
-            gainValue[i - 1] = engine.getValue("[Sampler" + [i] + "]",
-                "pregain");
-
-            if (gainValue[i - 1] <= 1) {
-                // increment value between 0 and 1
-                gainIncrement = 1 / 20; // 20 increments in one full knob turn   
-            } else {
-                // increment value between 1 and 4
-                gainIncrement = 3 / 20; // 20 increments in one full knob turn
-            }
+    if (deck.TapDown) {
+        engine.setValue('[BeatJump]', increase ? 'next': 'prev', true);
+        engine.setValue('[BeatJump]', increase ? 'next': 'prev', false);
+    }
 
-            // beat knobs sends 1 or 127 as value. If value = 127, turn is counterclockwise, we reduce gain
-            if (value === 127) {
-                gainIncrement = -gainIncrement;
-            }
+    if (deck.shiftKey) {
+        engine.setValue('[LoopMove]', increase ? 'next': 'prev', true);
+        engine.setValue('[LoopMove]', increase ? 'next': 'prev', false);
+    }
+};
 
-            gainValue[i - 1] = gainValue[i - 1] + gainIncrement;
+NumarkMixtrack3.OnBeatJump = function(value, group, control) {
+    if (!value) return;
+    // function to use to determine next jump size depends on value of control
+    var nextSizeFun = (control === 'next') ? nextJumpSize : prevJumpSize;
+    beatJumpSize = nextSizeFun(beatJumpSize);
+};
 
-            if ((gainValue[i - 1] + gainIncrement) < 0) {
-                gainValue[i - 1] = 0;
-            }
+NumarkMixtrack3.OnLoopMove = function(value, group, control) {
+    if (!value) return;
+    var nextSizeFun = (control === 'next') ? nextJumpSize : prevJumpSize;
+    loopMoveSize = nextSizeFun(loopMoveSize);
+};
 
-            if ((gainValue[i - 1] + gainIncrement) > 4) {
-                gainValue[i - 1] = 4;
-            }
+// when we hit a button on the UI that beat jumps by a certain amount, reset beatJumpSize
+// to reflect this. This mostly prevents the script and UI from becoming out of sync
+// when the script is reset but the UI is not.
+NumarkMixtrack3.OnBeatJumpX = function(value, group, control) {
+    if (!value) return;
+    beatJumpSize = parseInt(control.match(/\d/)[0]);        
+};
```

Comment:
I realized I should implement this in #1187. I already did for loop sizes, but didn't think to do so for the beatjump_X_forward/backward controls. So this shouldn't be necessary after that is merged.